src/modules/deal/deals.controller.ts

```diff
@@ -0,0 +1,30 @@
+import { Body, Controller, Post } from '@nestjs/common';
+import { ApiCreatedResponse, ApiInternalServerErrorResponse, ApiOperation } from '@nestjs/swagger';
+
+import { DealsService } from './deals.service';
+import { CreateDealDto } from './dto/deals-request.dto';
+import { CreateDealResponse } from './dto/deals-response.dto';
+
+@Controller('deals')
+export class DealsController {
+  constructor(private readonly dealsService: DealsService) {}
+
+  @Post()
+  @ApiOperation({ summary: 'Creates a new deal' })
+  @ApiCreatedResponse({
+    description: 'A deal has been successfully created',
+    type: CreateDealResponse,
+    isArray: false,
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred',
+  })
+  async createDeal(@Body() createDealDto: CreateDealDto): Promise<CreateDealResponse> {
+    const { headers } = await this.dealsService.createDeal(createDealDto);
+    // The createDeal call will return the location as a header, which we can then truncate the get the deal identifier
+    // The header value will look something like '/Portfolio/E1/Deal/0030596647', where the dealIdentifier is '0030596647'
+    // const dealIdentifier = headers['location']
+    return { dealIdentifier: createDealDto.dealIdentifier };
+    // Comment on the PR: is it okay to use the value in the request body as opposed to what is returned in the location header
```

Comment:
I assume the value in the request body will always be the same as the one from the header? 
if so, I think it will be best to use the one from the body - otherwise the format of the header may change in the future

Reply:
> I assume the value in the request body will always be the same as the one from the header?

This is the assumption I made, considering both variables are referred to as deal identifiers. It seemed strange that MuleSoft goes to the trouble of extracting the deal identifier from the Location header returned from ACBS, when a deal identifier is already included in the POST request body -- hence my uncertainty 😅

Reply:
Sounds good. 
@avaitonis  can you confirm that we can extract the dealIdentifier from the body and that nothing will break in the process?

Reply:
Hi,

In Mulesoft request body doesn't persist, unless it is saved to variable. And code tries return data from ACBS response when it is possible.

In this case Deal id in Location header will match Deal id in request body, happy to use id from body.